src/main/java/org/testsuite/SBMLTestSuiteRunnerWrapper.java

```diff
@@ -0,0 +1,237 @@
+
+        // Read the model and initialize solver
+        File sbmlfile = new File(filePath);
+        SBMLDocument document = (new SBMLReader()).readSBML(sbmlfile);
+        Model model = document.getModel();
+
+        // get timepoints
+        CSVImporter csvimporter = new CSVImporter();
+        MultiTable inputData = csvimporter.convert(model, resultsPath);
+        double[] timepoints = inputData.getTimePoints();
+        duration = timepoints[timepoints.length - 1]
+                - timepoints[0];
+
+        MultiTable solution;
+
+        // writes results to the output file in CSV format
+        File outputFile = new File(outputFilePath);
+        outputFile.createNewFile();
+
+        System.out.println(Paths.get(outputFilePath));
+        FileWriter csvWriter = new FileWriter(outputFilePath);
+
+        StringBuilder output = new StringBuilder("");
+
+        if (model.getExtension(FBCConstants.shortLabel) != null) {
+
+            FluxBalanceAnalysis solver = new FluxBalanceAnalysis(document);
+            if (solver.solve()) {
+                Map<String, Double> fbcSolution = solver.getSolution();
+
+                System.out.println(fbcSolution);
+
+                BufferedReader reader = new BufferedReader(new FileReader(resultsPath));
+                String[] keys = reader.readLine().trim().split(",");
+
+                for (int i = 0; i < keys.length - 1; i++) {
+                    output.append(keys[i]).append(",");
+                }
+                output.append(keys[keys.length - 1]).append("\n");
+
+                for (String key : keys) {
+                    output.append(fbcSolution.get(key)).append(",");
+                }
+
+                if (output.length() > 0) {
+                    output.deleteCharAt(output.length() - 1);
+                    output.append("\n");
+                }
+            }
+
+        } else {
+            if (model.getExtension(CompConstants.shortLabel) == null) {
+                DESSolver solver = new RosenbrockSolver();
+                solver.setStepSize(duration / steps);
+
+                if (solver instanceof AbstractDESSolver) {
+                    solver.setIncludeIntermediates(false);
+                }
+
+                if (solver instanceof AdaptiveStepsizeIntegrator) {
+                    ((AdaptiveStepsizeIntegrator) solver).setAbsTol(TOLERANCE_FACTOR * absolute);
+                    ((AdaptiveStepsizeIntegrator) solver).setRelTol(TOLERANCE_FACTOR * relative);
+                }
+
+                SBMLinterpreter interpreter = new SBMLinterpreter(model, 0, 0, 1, amountHash);
+
+                // Compute the numerical solution of the problem
+                solution = solver.solve(interpreter, interpreter.getInitialValues(), timepoints);
+            } else {
+                CompSimulator compSimulator = new CompSimulator(sbmlfile);
+                double stepSize = (duration / steps);
+
+                solution = compSimulator.solve(stepSize, duration);
```

Comment:
If we will not provide any changes to the CompSimulator one needs to deal with exceptions here (we are throwing except. in the corresponding solve method). Please consider this.